service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -439,7 +439,10 @@ public synchronized void init(HiveConf hiveConf) {
     }
 
     // Add a shutdown hook for catching SIGTERM & SIGINT
-    ShutdownHookManager.addShutdownHook(() -> graceful_stop());
+    long timeout = HiveConf.getTimeVar(getHiveConf(),
+        HiveConf.ConfVars.HIVE_SERVER2_GRACEFUL_STOP_TIMEOUT, TimeUnit.SECONDS);
+    // Add extra time for releasing the resources
+    ShutdownHookManager.addGracefulShutDownHook(() -> graceful_stop(), timeout + 30);
```

Comment:
what is this extra 30 seconds for? The default is 1800s. What would an extra 30 seconds do anything different. Seems like a small percentage of the default.

Reply:
The extra seconds are for HiveServer2 calling stop(), which in turn releasing resources, such as cancelling the running operations, cleaning local files, etc.
Without it, the shutdown scripts will issue 'kill -9' to kill the HiveServer2 directly when timeout, so the HiveServer2 may have no chance to call stop...